protocols/gossipsub/src/behaviour.rs

```diff
@@ -83,28 +84,59 @@ pub struct Gossipsub {
     /// Message cache for the last few heartbeats.
     mcache: MessageCache,
 
-    // We keep track of the messages we received (in the format `string(source ID, seq_no)`) so that
-    // we don't dispatch the same message twice if we receive it twice on the network.
-    received: LruCache<MessageId, ()>,
-
     /// Heartbeat interval stream.
     heartbeat: Interval,
+
+    /// The peer_id public to add to messages if signing is enabled and the current libp2p key is
+    /// not inlined in the `PeerId`.  
+    key: Option<Vec<u8>>,
 }
 
 impl Gossipsub {
     /// Creates a `Gossipsub` struct given a set of parameters specified by `gs_config`.
-    pub fn new(local_peer_id: PeerId, gs_config: GossipsubConfig) -> Self {
-        let local_peer_id = if gs_config.no_source_id {
+    pub fn new(keypair: Keypair, gs_config: GossipsubConfig) -> Self {
+        // Set up the router given the configuration settings.
+
+        // Sets up the source_id of published messages.
+        let message_source_id = if gs_config.no_source_id {
             PeerId::from_bytes(crate::config::IDENTITY_SOURCE.to_vec()).expect("Valid peer id")
```

Comment:
I can't find the concept of the identity source id in the specification. Could you point me to it?

Reply:
This was something I added as extra pseudo-privacy. Since this addition go has introduced an "author" which can be modified. 

This has since been removed in favour of the `Signing` enum, which when sent to disabled gives you the ability to set the `author`. 

The identity was just an arbitrary peer ID i was using, which is easily identifiable.